# Graded assignment 1

Nick Starke (2005453)


# Packages

We need the package `tidyverse`

In [101]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [102]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.1295136 0.5756212 0.6080274 0.4536431 0.5120856 0.9878451 0.5743630
2 0.8025518 0.9051170 0.8702453 0.5138387 0.2098664 0.5981043 0.7309977
  V8         V9        V10        V11       V12       V13        V14       
1 0.08925122 0.5509237 0.57640300 0.5956547 0.4272944 0.06227968 0.74997376
2 0.44355683 0.8772727 0.01157267 0.9452469 0.3274702 0.33494520 0.03842414
  V15       V16       V17       V18       V19       V20      
1 0.4662170 0.5684109 0.9264835 0.3142512 0.3821760 0.4647166
2 0.7407246 0.6728209 0.5521671 0.6188755 0.9789705 0.7252601

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [103]:
df2 <- df1 %>% gather(Key, Value)

head(df2)

Key Value    
1 V1  0.1295136
2 V1  0.8025518
3 V2  0.5756212
4 V2  0.9051170
5 V3  0.6080274
6 V3  0.8702453

## Assignment 2

We will generate some data:

In [104]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [105]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [106]:
df5 <- merge(df3, df4, all= TRUE)
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook